types.d.ts

```diff
@@ -195,6 +195,13 @@ export interface SimpleSupportStatement {
      */
     value_to_set?: string;
   }[];
+
+  /**
+   * An optional changeset URL or commit URL for the change which implemented the feature in the
+   * source code for the associated browser; e.g. a https://trac.webkit.org/changeset/ URL.
+   */
+  implementation_url?: string;
```

Comment:
Should an array be allowed, because I see some notes with two tracking bugs?

https://github.com/mdn/browser-compat-data/blob/aefff297d4f0ef9ba8132b626f8882254345e852/api/AudioParam.json#L260

Reply:
> Should an array be allowed, because I see some notes with two tracking bugs?

We could allow an array (as we do already for `spec_url`) — but the reason I restricted it to a single string in the initial patch here is that I was not aware for any known cases where we’d need to specify more than one URL.

> https://github.com/mdn/browser-compat-data/blob/aefff297d4f0ef9ba8132b626f8882254345e852/api/AudioParam.json#L260

OK, looking at that case, I see the subject of https://bugzil.la/1171438 is *“Implement the new behavior for AudioParam.value”*, so that’s clearly a (yet-unimplemented) feature/implementation request — while looking at https://bugzil.la/1567777 see that one looks like it’s just a bug report (and one which is actually marked as a duplicate of the https://bugzil.la/1171438 feature request…)

So to me at least, that’s not a case that demonstrates a concrete need for allowing an array for the `implemenation_url` value — because the https://bugzil.la/1567777 URL isn’t a changeset URL at all, and neither actually is the https://bugzil.la/1171438, since the feature hasn’t actually been implemented yet.

Reply:
Cool, but that's just a one example. Searching by `https://[^d].+https://[^d]` gives 61 results, so probably something to consider.

Reply:
> Cool, but that's just a one example. Searching by `https://[^dw].+https://[^dw]` gives 43 results, so probably something to consider.
> 
> Edit: Those may also turn out as just bug reports, though. A short skimming couldn't find a valid example.

Right. But you’ve make me realize that for Firefox features, rather than an https://bugzil.la URL, what we’d instead want to have in `implementation_url` values is a https://hg.mozilla.org/mozilla-central/rev URL — for example:

https://hg.mozilla.org/mozilla-central/rev/8213cdcc0378

The subject of that changeset is *“Implement FetchEvent.handled”* and it happens to have a corresponding bug at https://bugzilla.mozilla.org/show_bug.cgi?id=1617822 but it seems clear that we’d want to cite the https://hg.mozilla.org/mozilla-central/rev/8213cdcc0378 URL instead — and so, to generalize from that, we’d want to cite https://hg.mozilla.org/mozilla-central/rev URLs rather than https://bugzil.la or https://bugzilla.mozilla.org URLs.

And then to generalize even further, I think regardless of which browser it’s for, we’d want to restrict the value to being a URL for the browser project’s revision log rather than for the project’s bug tracker.

So that means we’d want either a https://trac.webkit.org/changeset/, https://hg.mozilla.org/mozilla-central/rev/, or 
https://chromium.googlesource.com/chromium/src or https://crrev.com/ URL.

Given that, I’ve pushed another commit here that updates the docs part of patch to explicitly mention those — and I’ve also updated the schema change itself to explicitly limit the value to a pattern that matches URLs in one of those forms.